Source/Core/ClippingPlane.js

```diff
@@ -0,0 +1,155 @@
+            },
+            set : function(value) {
+                //>>includeStart('debug', pragmas.debug);
+                Check.typeOf.object('value', value);
+                //>>includeEnd('debug');
+                if (defined(this.onChangeCallback) && !Cartesian3.equals(this._normal._cartesian3, value)) {
+                    this.onChangeCallback(this.index);
+                }
+                // Set without firing callback again
+                Cartesian3.clone(value, this._normal._cartesian3);
+            }
+        }
+    });
+
+    /**
+     * Create a ClippingPlane from a Plane object.
+     *
+     * @param {Plane} plane The plane containing parameters to copy
+     * @param {ClippingPlane} [result] The object on which to store the result
+     * @returns {ClippingPlane} The ClippingPlane generated from the plane's parameters.
+     */
+    ClippingPlane.fromPlane = function(plane, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('plane', plane);
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new ClippingPlane(plane.normal, plane.distance);
+        } else {
+            result.normal = plane.normal;
+            result.distance = plane.distance;
+        }
+        return result;
+    };
+
+    /**
+     * Clones the ClippingPlane without setting its ownership.
+     * @param {ClippingPlane} clippingPlane The ClippingPlane to be cloned
+     * @param {ClippingPlane} [result] The object on which to store the cloned parameters.
+     * @returns {ClippingPlane} a clone of the input ClippingPlane
+     */
+    ClippingPlane.clone = function(clippingPlane, result) {
+        if (!defined(result)) {
+            return new ClippingPlane(clippingPlane.normal, clippingPlane.distance);
+        }
+        result.normal = clippingPlane.normal;
+        result.distance = clippingPlane.distance;
+        return result;
+    };
+
+    /**
+     * Wrapper on Cartesian3 that allows detection of Plane changes from "members of members," for example:
+     *
+     * var clippingPlane = new ClippingPlane(...);
+     * clippingPlane.normal.z = -1.0;
+     *
+     * @private
+     */
+    function UpdateChangedCartesian3(normal, clippingPlane) {
+        this._clippingPlane = clippingPlane;
+        this._cartesian3 = Cartesian3.clone(normal);
+    }
+
+    function makeGetterStter(key) {
+        return {
+            get : function() {
+                return this._cartesian3[key];
+            },
+            set : function(value) {
+                //>>includeStart('debug', pragmas.debug);
+                Check.typeOf.number('value', value);
+                //>>includeEnd('debug');
+                if (defined(this._clippingPlane.onChangeCallback) && value !== this._cartesian3[key]) {
+                    this._clippingPlane.onChangeCallback(this._clippingPlane.index);
+                }
+                this._cartesian3[key] = value;
```

Comment:
I'm sure everyone is aware that this switches the object instance into dictionary mode with slower member lookups.  Totally fine for the expected use cases here.  I wonder - but do not know for sure - if this also impacts the JIT's ability to optimize other paths using Cartesian3's, e.g., if these Cartesian3's are passed to other functions that were optimized to have a fast path with them - is that rolled back or a slow path is duplicated?  No action required, just curious and I'm sure we will find out if it is an issue...